src/main/java/frc/robot/subsystems/PoseEstimation.java

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems;
+
+import java.io.IOException;
+import edu.wpi.first.apriltag.AprilTagFieldLayout;
+import edu.wpi.first.apriltag.AprilTagFields;
+import edu.wpi.first.math.estimator.SwerveDrivePoseEstimator;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.wpilibj.smartdashboard.Field2d;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.SwerveConfig;
+import frc.robot.constants.VisionConstants;
+
+public class PoseEstimation extends SubsystemBase {
+  
+  // Subsystems
+  SwerveSubsystem swerve;
+  Limelight lime;
+
+  // Fields
+  AprilTagFieldLayout layout;
+  Field2d field;
+
+  // Estimator
+  SwerveDrivePoseEstimator poseEstimator;
+  public PoseEstimation(SwerveSubsystem swerve, Limelight limelight) {
+    this.swerve = swerve;
+    this.lime = limelight;
+
+    try {
+      layout = AprilTagFieldLayout.loadFromResource(AprilTagFields.k2023ChargedUp.m_resourceFile);
+    } catch (IOException e) {
+      e.printStackTrace();
```

Comment:
When does this happen? How does the software recover after this exception is caught?